### PR TITLE
feat(celery): Allow to override propagate_traces per task

### PR DESCRIPTION
**Pull request by jan-auer**
Adds support for a `sentry-propagate-traces` header on `apply_async` that
overrides the default behavior set through the Celery integration's
`propagate_traces` flag.

Example usage:

```py
my_task.apply_async(..., headers={"sentry-propagate-traces": False})
```

Example use case:
We ourselves have a task that is running once every two weeks and it is sampled with ~0.01 Percent. So we can one transaction from this task a year (give or take). This task starts hundreds of child tasks. All those child tasks will inherit the sampling decision from the original task and thus will be dropped most of the times. But we want to have those child tasks transactions in our backend no matter the sampling decision of the parent.

### CODE REVIEW

**Comment by antonpirker** [comment]
Documentation PR: https://github.com/getsentry/sentry-docs/pull/7757

File: sentry_sdk/integrations/celery.py [hunk 1]
```diff
@@ -140,13 +140,21 @@ def apply_async(*args, **kwargs):
         # type: (*Any, **Any) -> Any
         hub = Hub.current
         integration = hub.get_integration(CeleryIntegration)
-        if integration is not None and integration.propagate_traces:
+
```
Comment (sentrivana) [inline sentry_sdk/integrations/celery.py:143]: We could just early return here if integration is None and save ourselves any future `integration is not None` checks below:
```suggestion
if integration is None:
    return f(*args, **kwargs)
```

File: sentry_sdk/integrations/celery.py [hunk 2]
```diff
@@ -140,13 +140,21 @@ def apply_async(*args, **kwargs):
...
+        # Note: kwargs can contain headers=None, so no setdefault!
+        # Unsure which backend though.
+        kwarg_headers = kwargs.get("headers") or {}
+        propagate_traces = kwarg_headers.pop("sentry-propagate-traces", None)
+        if propagate_traces is None and integration is not None:
```
Comment (sentrivana) [inline sentry_sdk/integrations/celery.py:148]: If you go with my previous suggestion:
```suggestion
        if propagate_traces is None:
```

File: sentry_sdk/integrations/celery.py [hunk 3]
```diff
@@ -140,13 +140,21 @@ def apply_async(*args, **kwargs):
...
+            propagate_traces = integration.propagate_traces
+
+        if propagate_traces:
             with hub.start_span(
                 op=OP.QUEUE_SUBMIT_CELERY, description=args[0].name
             ) as span:
                 with capture_internal_exceptions():
                     headers = dict(hub.iter_trace_propagation_headers(span))
-                    if integration.monitor_beat_tasks:
+                    if integration is not None and integration.monitor_beat_tasks:
```
Comment (sentrivana) [inline sentry_sdk/integrations/celery.py:157]: Same, if you go with the original suggestion:
```suggestion
                    if integration.monitor_beat_tasks:
```

**Review by sentrivana — APPROVED** [review]
Lgtm, left a small suggestion

File: sentry_sdk/integrations/celery.py (hunk 1 above)
Comment (antonpirker) [reply sentry_sdk/integrations/celery.py:143]: YES!

File: sentry_sdk/integrations/celery.py (hunk 3 above)
Comment (antonpirker) [reply sentry_sdk/integrations/celery.py]: done

File: sentry_sdk/integrations/celery.py (hunk 2 above)
Comment (antonpirker) [reply sentry_sdk/integrations/celery.py]: done

**Comment by antonpirker** [comment]
Diff looks now ugly, but code is way more readable! 💯

**Pull request merged**